documentation/modules/cruise-control/con-optimization-proposals.adoc

```diff
@@ -0,0 +1,91 @@
+
+Use the summary (shown in the `Status` of the `KafkaRebalance` resource) to decide whether to implement the cluster rebalance, or change the optimization goals.
+
+NOTE: All optimization proposals are dry runs: you cannot implement a cluster rebalance without first generating an optimization proposal. 
+There is no limit to the number of optimization proposals that can be generated.
+
+[discrete]
+== Cached optimization proposal
+
+Cruise Control maintains a _cached optimization proposal_ based on the configured default optimization goals.
+Generated from the workload model, the cached optimization proposal is updated every 15 minutes to reflect the current state of the Kafka cluster.
+If you generate an optimization proposal using the default optimization goals, Cruise Control returns the most recent cached proposal.
+
+To change the cached optimization proposal refresh interval, edit the `proposal.expiration.ms` setting in the Cruise Control deployment configuration.
+Consider a shorter interval for fast changing clusters, although this increases the load on the Cruise Control server.
+
+[id="contents-optimization-proposals"]
+[discrete]
+== Contents of optimization proposals
+
+The following table explains the properties contained in an optimization proposal:
+
+[cols="2*",options="header",stripes="none",separator=¦]
+|===
+
+m¦JSON property
+¦Description
+
+m¦numIntraBrokerReplicaMovements
+¦The total number of partition replicas that will be transferred between the disks of the cluster's brokers.
+
+*Performance impact during rebalance operation*: Relatively high, but lower than `numReplicaMovements`.
+
+m¦excludedBrokersForLeadership
+¦Not yet supported. An empty list is returned.
+
+m¦numReplicaMovements
+¦The number of partition replicas that will be moved between separate brokers.
+
+*Performance impact during rebalance operation*: Relatively high.
+
+m¦onDemandBalancednessScoreBefore, onDemandBalancednessScoreAfter
+¦A measurement of the overall _balancedness_ of a Kafka Cluster, before and after the optimization proposal was generated.
+
+The calculation is 100 minus the sum of the `BalancednessScore` of each violated soft goal. Cruise Control assigns a `BalancednessScore` to every optimization goal based on several factors, including priority (the goal's position in the list of `default.goals` or user-provided goals).
+
+The `Before` score is based on the current configuration of the Kafka cluster.
+The `After` score is based on the generated optimization proposal.
+
+m¦intraBrokerDataToMoveMB
+¦The sum of the size of each partition replica that will be moved between disks on the same broker (see also `numIntraBrokerReplicaMovements`).
+
+m¦recentWindows
+¦The number of metrics windows upon which the optimization proposal is based.
+
+m¦dataToMoveMB
+¦The sum of the size of each partition replica that will be moved to a separate broker (see also `numReplicaMovements`).
```

Comment:
I guess it might be worth calling out that is (and to a lesser extent `intraBrokerDataToMoveMB`) also provide a guide about how costly the balancing is going to be.

Reply:
That is a good idea.

Reply:
Sorry, I don't quite understand your suggestion - @tombentley

Reply:
The bigger this number is (and to a lesser extent `intraBrokerDataToMoveMB`), the longer a rebalance will take .